lib/libzfs/libzfs_import.c

```diff
@@ -209,14 +203,14 @@ zpool_clear_label(int fd)
 		}
 	}
 
-	/* Clear the L2ARC header. */
-	if (clear_l2arc_header &&
-	    pwrite64(fd, l2dhdr, sizeof (l2arc_dev_hdr_phys_t),
-		VDEV_LABEL_START_SIZE) == sizeof (l2arc_dev_hdr_phys_t))
+	if (clear_l2arc_header) {
+		memset(label, 0, sizeof (l2arc_dev_hdr_phys_t));
+		if (pwrite64(fd, label, sizeof (l2arc_dev_hdr_phys_t),
+		    VDEV_LABEL_START_SIZE) == sizeof (l2arc_dev_hdr_phys_t))
 			header_cleared = B_TRUE;
+	}
```

Comment:
Even though pretty much a certainty that the `vdev_label_t` struct will always be bigger than the `l2arc_dev_hdr_phys_t`structure it wouldn't be a bad idea to `assert` that here.

Reply:
Rebased; `_Static_assert` added